### PR TITLE
Separate eval from train for lambda_i and ci_ma

### PR DESCRIPTION
**Pull request by chokkyvista**
1. only update lambda **during training**
2. use **separate** moving-averages for `train` vs. `eval` (similar to batch-norm I guess?)

I find 1 to be **crucial** for stabilizing the training under GECO - without it I can only train the "latent" model with the default batch size (i.e. 256) regardless of how hard I tune the learning rate.
Yet with it fixed, I've managed to train with much larger batch sizes, e.g. 1024.

_(edit: correct legend color)_
 batch_size=256 (default) | batch_size=1024 (4x)
:-------------------------:|:-------------------------:
image | image

### CODE REVIEW

**Comment by chokkyvista** [comment]
(no code-change in the above [force-push](https://github.com/bastings/interpretable_predictions/compare/1307d75f3b76e9b0f44f737e9b0af36348d4ce80..85e416631c61337a2aa413ce9ed7795d0d976951), only switching `user.email` to match my github account)
